site/src/pages/api/uploadUserAvatar.api.ts

```diff
@@ -0,0 +1,50 @@
+import { uploadFileStreamToS3 } from "../../lib/awsS3";
+import {
+  AuthenticatedApiRequest,
+  createAuthenticatedHandler,
+} from "../../lib/handler/authenticatedHandler";
+
+import {
+  FormRequestExtensions,
+  imageUploadMultipart,
+} from "../../lib/middleware/imageUploadMultipart";
+
+type Response = {
+  fullUrl: string;
+};
+
+export default createAuthenticatedHandler<
+  AuthenticatedApiRequest & FormRequestExtensions,
+  Response
+>()
+  .use(imageUploadMultipart)
+  .post(async (req, res) => {
+    const { body, user, db } = req;
+
+    if (body?.upload && body.upload?.file) {
+      const { file, stream } = body.upload;
+      const { fullUrl, s3Key } = await uploadFileStreamToS3(
+        stream,
+        file.mimetype || "",
+        new Date().valueOf().toString(),
+        `avatars/${user.id}`,
+        "image",
+      );
+
+      // @todo should we delete previous user avatar before replacing it?
+
+      await user.update(db, {
+        userAvatar: { url: fullUrl, s3Key },
+      });
+
+      res.status(200).json({ fullUrl });
+    } else {
+      throw new Error("No file given. Please upload a file.");
```

Comment:
There's a `formatErrors` function that standardises how errors are returned - we should give them all status codes too (where anticipated / user error) so they don't come through as `500`:
```suggestion
      res.status(400).json(
        formatErrors({
          msg: "No file given. Please upload a file.",
          param: "image",
        }),
      );
```